daemon/graphdriver/zfs/zfs.go

```diff
@@ -361,6 +361,17 @@ func (d *Driver) Remove(id string) error {
 	name := d.zfsPath(id)
 	dataset := zfs.Dataset{Name: name}
 	err := dataset.Destroy(zfs.DestroyRecursive)
+	if err != nil {
+		var errZfs *zfs.Error
+		isZfsError := errors.As(err, &errZfs)
+		if isZfsError && strings.HasSuffix(errZfs.Stderr, "dataset does not exist\n") {
```

Comment:
I'm slightly on the fence if we should consider stripping the newline from `Stderr`. Perhaps we should; I just had a quick peek at the zfs code and it already does some manipulation of the output it got, so it may be a bit brittle if we depend on the newline to be present.

Let me add a quick `strings.TrimSpace` here to prevent formatting changes from breaking this.